browser/js/components/Comment/index.js

```diff
@@ -0,0 +1,207 @@
+  // solutionCodeLink: "http://www.google.com",
+  // tags: ['foo', 'bar', 'baz'],
+  // attachments: ['foo', 'bar', 'baz'],
+  // annotation: null,
+  criteria: criteria,
+  markdown: "blah blah blah"
+}
+
+class Comment extends Component {
+  constructor(props){
+    super(props);
+    this.buttonOnClickHandler = this.editMode.bind(this);
+    this.onClickDoneHandler = this.editModeDone.bind(this);
+  }
+  componentWillReceiveProps(nextProps){
+    console.log('received props', nextProps);
+    if(nextProps.contents) {
+      let { contents } = nextProps;
+      this.contents = contents;
+    }
+  }
+  renderComment () {
+    console.log('rendering comment');
+    let contents = this.props.contents || defaultContents ;
+    let isEditing = this.props.isEditing;
+    let buttonStyle = styles.assessmentButtons;
+    return ( <div>
+      { (!contents.description && isEditing) ? <RaisedButton style={buttonStyle} label="Add Description" /> : "" }
+      { (!contents.score && isEditing) ? <RaisedButton style={buttonStyle} label="Add Score" /> : ""}
+      { (!contents.solutionCodeLink && isEditing) ? <RaisedButton style={buttonStyle} label="Add Solution Code" /> : ""}
+      { (!contents.tags && isEditing) ? <RaisedButton style={buttonStyle} label="Add Tag" /> : ""}
+      { (!contents.attachments && isEditing) ? <RaisedButton style={buttonStyle} label="Add Attachment" /> : ""}
+      { (!contents.selection) ? (isEditing ? <RaisedButton style={buttonStyle} label="Add Annotation" /> : "") : (
+        <div>
+        <div>
+          <TextField
+          hintText="Annotate Code"
+          defaultValue = {contents.selection.toString()}
+          floatingLabelText="Code Annotation"
+          multiLine={true}
+           />
+          <br />
+        </div>
+        < FlatButton href="/grade">Go to Code</ FlatButton>
+        </div>
+      ) }
+      { (!contents.criteria && isEditing) ? <RaisedButton style={buttonStyle} label="Add Criteria" /> : contents.criteria }
+      { (!contents.markdown && isEditing) ? <RaisedButton style={buttonStyle} label="Add Markdown" /> : (
+        <TextField
+          hintText="Add a comment"
+          defaultValue = {contents.markdown}
+          floatingLabelText="Instructor Comments"
+          multiLine={true}
+           />
+      ) }
+
+      </div>)
+
+  }
+  editMode(){
+    this.props.dispatch({type: 'COMMENT_EDIT_START', payload: {key: this.props.commentIndex}})
+  }
+  editModeDone(){
+    this.props.dispatch({type: 'COMMENT_EDIT_DONE', payload: {key: null}})
+  }
+  render(){
+    const iconButtonElement = (
+      <IconButton
+        touch={true}
+        tooltip="edit comment"
+        tooltipPosition="bottom-left"
+        onClick={this.buttonOnClickHandler}
+      >
+        <ModeEdit color={grey400} />
+      </IconButton>
+    );
+
+    const rightIconMenu = (
+      <IconMenu iconButtonElement={iconButtonElement}>
+        <MenuItem>Reply</MenuItem>
+        <MenuItem>Forward</MenuItem>
```

Comment:
I am not sure exactly what you guys are thinking, but back is usually good if you have a forwards. Just as a UX thought